File: Homework_1/StatisticsFirstAssignmentCSharp/StatisticsFirstAssignmentCSharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace StatisticsFirstAssCSharp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void WriteButton_Click(object sender, EventArgs e)
        {
            this.richTextBox1.Text += "Hello World!";
        }

        private void CleanButton_Click(object sender, EventArgs e)
        {
            this.richTextBox1.Text = "";
        }

        private void ChangeColor_MouseEnter(object sender, EventArgs e)
        {
            this.richTextBox1.BackColor = System.Drawing.Color.Aqua;
        }

        private void ChangeColor_MouseLeave(object sender, EventArgs e)
        {
            this.richTextBox1.BackColor = System.Drawing.Color.White;
        }

        private void WriteButton_DragEnter(object sender, System.Windows.Forms.DragEventArgs e)
        {
            if (e.Data.GetDataPresent(DataFormats.FileDrop))
                e.Effect = DragDropEffects.All;
            else
                e.Effect = DragDropEffects.None;
        }

        private void WriteButton_DragDrop(object sender, System.Windows.Forms.DragEventArgs e)
        {
            string[] s = (string[])e.Data.GetData(DataFormats.FileDrop, false);
            int i;
            for (i = 0; i < s.Length; i++)
                this.richTextBox1.Text += s[i] + "\n";
        }
    }
}

File: Homework_1/StatisticsFirstAssignmentVB/StatisticsFirstAssignmentVB/Form1.vb
Imports System.IO

Public Class Form1

    Private Sub WriteButton_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Me.RichTextBox1.Text = "Hello World "
    End Sub

    Private Sub CleanButton_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Me.RichTextBox1.Text = ""
    End Sub

    Private Sub ChangeColor_MouseEnter(sender As Object, e As EventArgs) Handles RichTextBox1.MouseEnter
        Me.RichTextBox1.BackColor = System.Drawing.Color.Aqua
    End Sub

    Private Sub ChangeColor_MouseLeave(sender As Object, e As EventArgs) Handles RichTextBox1.MouseLeave
        Me.RichTextBox1.BackColor = System.Drawing.Color.White
    End Sub

    Private Sub WriteButton_DragEnter(sender As Object, e As DragEventArgs) Handles Button1.DragEnter
        If e.Data.GetDataPresent(DataFormats.FileDrop) Then
            e.Effect = DragDropEffects.All
        Else
            e.Effect = DragDropEffects.None
        End If
    End Sub

    Private Sub WriteButton_DragDrop(sender As Object, e As DragEventArgs) Handles Button1.DragDrop
        Dim files() As String = e.Data.GetData(DataFormats.FileDrop, False)
        For Each path In files
            Me.RichTextBox1.Text += path + vbCrLf
        Next
    End Sub

End Class
